census_extractomatic/api.py

```diff
@@ -2009,6 +2021,303 @@ def table_geo_comparison_rowcount(table_id):
 
     return resp
 
+## COMBINED LOOKUPS ##
+
+@app.route("/2.1/full-text/search")
+@qwarg_validate({
+    'q':   {'valid': NonemptyString()},
+    'type': {'valid': OneOf(allowed_searches), 'default': allowed_searches[3]},
+})
+@crossdomain(origin='*')
+def full_text_search():
+
+    def do_search(db, q, object_type):
+        """ Search for objects (profiles, tables, topics) matching query q.
+
+        Return a list, because it's easier to work with than a SQLAlchemy 
+        ResultProxy object (notably, the latter does not support indexing).
+        """
+
+        q_profiles = """SELECT text1 AS display_name, 
+                               text2 AS sumlevel,
+                               text3 AS sumlevel_name,
+                               text4 AS full_geoid,
+                               text5 AS population, 
+                               text6 AS priority,
+                               ts_rank(document, to_tsquery('simple', '{0}')) AS relevance,
```

Comment:
Rather than use Python's string formatting here, let's use the built-in string replacement stuff that SQLAlchemy has. [This is a fairly good article describing in more detail what I'm talking about](https://www.quantifiedcode.com/knowledge-base/security/Prevent%20SQL%20injections%20by%20avoiding%20string%20interpolations/3cObiWuF).